agent/api/container.go

```diff
@@ -95,6 +107,9 @@ type Container struct {
 	// 'Internal' containers are ones that are not directly specified by
 	// task definitions, but created by the agent
 	IsInternal bool
+	// InternalContainerType represents the internal container type when
+	// IsInternal is set to true
+	InternalContainerType InternalContainer
```

Comment:
Should be addressed in the next commit that I'm going to push, where `IsInternal` has been refactored as an enum.